src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,181 @@
+        this.sections = sections;
+    }
+
+    public static Sections valueOf(List<Section> sections) {
+        return new Sections(sections);
+    }
+
+    public void addSection(Section newSection) {
+        if (!isSectionsEmpty()) {
+            validateNotAddedSection(newSection);
+            validateContainsUpStationOrDownStation(newSection);
+            update(newSection);
+        }
+        sections.add(newSection);
+    }
+
+    private boolean isSectionsEmpty() {
+        return sections.isEmpty();
+    }
+
+    private void validateNotAddedSection(Section section) {
+        if (containsUpStationAndDownStation(section)) {
+            throw new IllegalArgumentException("이미 등록된 구간 입니다.");
+        }
+    }
+
+    private boolean containsUpStationAndDownStation(Section section) {
+        return isUpStationExisted(section) && isDownStationExisted(section);
+    }
+
+    private void validateContainsUpStationOrDownStation(Section section) {
+        if (containsNoneOfUpStationAndDownStation(section)) {
+            throw new IllegalArgumentException("등록할 수 없는 구간 입니다.");
+        }
+    }
+
+    private boolean containsNoneOfUpStationAndDownStation(Section section) {
+        return !isUpStationExisted(section) && !isDownStationExisted(section);
+    }
+
+    private boolean isUpStationExisted(Section section) {
+        return orderedStations().stream().anyMatch(station -> station.equals(section.upStation()));
+    }
+
+    private boolean isDownStationExisted(Section section) {
+        return orderedStations().stream().anyMatch(station -> station.equals(section.downStation()));
+    }
+
+    public List<Station> orderedStations() {
+        Map<Station, Station> section = sections.stream()
+                .collect(Collectors.toMap(Section::upStation, Section::downStation));
+        Station downStation = findUpStation();
+        List<Station> stations = new ArrayList<>();
+        stations.add(downStation);
+        while (section.containsKey(downStation)) {
+            stations.add(section.get(downStation));
+            downStation = section.get(downStation);
+        }
+        return stations;
+    }
+
+    private void update(Section newSection) {
+        sections.forEach(section -> section.update(newSection));
+    }
+
+    public void deleteSection(Station station) {
+        validateNumberOfSections();
+        if (isEqualToFirstUpStation(station)) {
+            sections.remove(findSectionWithSameUpStation(station));
+            return;
+        }
+        if (isEqualToLastDownStation(station)) {
+            sections.remove(findSectionWithSameDownStation(station));
+            return;
+        }
+        Section nextSection = findSectionWithSameUpStation(station);
+        Section prevSection = findSectionWithSameDownStation(station);
+        sections.remove(nextSection);
+        sections.remove(prevSection);
+        sections.add(mergeSection(prevSection, nextSection));
```

Comment:
```suggestion
        Section nextSection = findSectionWithSameUpStation(station);
        Section prevSection = findSectionWithSameDownStation(station);
        sections.remove(nextSection);
        sections.remove(prevSection);
        sections.add(mergeSection(prevSection, nextSection));
```

의도를 들어내는 메서드로 분리한다면, 해당 로직이 어떤 역할을 하는지 이해하기 쉽지않을까 생각되는데,
어떻게 생각하시나요~! :thinking: